### PR TITLE
Linearity of Expectation

### PR DESCRIPTION
**Pull request by HaccerKat**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

Yay linearity of expectation :)

### CODE REVIEW

File: content/4_Gold/Combinatorics.mdx [hunk 1]
```diff
@@ -934,6 +934,76 @@ int32_t main() {
...
+}
+
+int32_t main() {
+	std::ios::sync_with_stdio(false);
+	cin.tie(NULL);
+	solve();
+}
+```
+
 </CPPSection>
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:1007]: uh
the python solution for marbles is right below here
u sure this is int ehr gith place?

File: content/4_Gold/Combinatorics.mdx [hunk 2]
```diff
@@ -868,28 +868,41 @@ non-empty. Thus, the answer is obtained by the second formula: $\binom{n-1}{k-1}
...
-	long long ret = 1;
-	for (int i = 0; i < k; i++) {
-		// this is done instead of *= for divisibility issues
-		ret = ret * (n - i) / (i + 1);
-	}
-	return ret;
+	long long ans = 1;
+	for (int i = 0; i < k; i++) { ans = ans * (n - i) / (i + 1); }
+	return ans;
 }
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:878]: uh why'd you undo the changes

Comment (HaccerKat) [reply content/4_Gold/Combinatorics.mdx:878]: not sure where that came from

Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.mdx:878]: kk i'll fix it myself sometime

Comment (HaccerKat) [reply content/4_Gold/Combinatorics.mdx:878]: the bot is also doing stuff like that

Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.mdx:878]: stuff like what

Comment (HaccerKat) [reply content/4_Gold/Combinatorics.mdx:878]: like that for loop changing from multiple lines to one

maybe i should pull code every time i edit

Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.mdx:878]: no yeah we just have to live with that

> maybe i should pull code every time i edit

yeah this is also good

**Review by SansPapyrus683 — APPROVED** [review]

File: content/4_Gold/Combinatorics.mdx [hunk 3]
```diff
@@ -903,49 +920,174 @@ For example, let $X$ be the probability distribution of a fair 6-sided die. $P(X
 
 ### Explanation
 
-Let $X$ represent the probability distribution of maximum number of candies a child gets. To get $E[X]$, we just need $P(X = x)$ for $1 \leq x \leq k$. $(\frac{x}{k})^n$ is the probability that each child gets at most $x$ candies. To get $P(X = x)$, we must subtract out the probability that each child gets strictly less than $x$ candies, which is $(\frac{x - 1}{k})^n$.
+Let $X$ represent the probability distribution of maximum number of candies a child gets. To get $E[X]$, we  need $P(X = x)$ for $1 \leq x \leq k$. $(\frac{x}{k})^n$ is the probability that each child gets at most $x$ candies. To get $P(X = x)$, we must subtract out the probability that each child gets strictly less than $x$ candies, which is $(\frac{x - 1}{k})^n$.
```
Comment (ryanchou-dev) [inline content/4_Gold/Combinatorics.mdx:923]:
```suggestion
Let $X$ represent the probability distribution of the maximum number of candies a child gets. To get $E[X]$, we need $P(X = x)$ for $1 \leq x \leq k$. $(\frac{x}{k})^n$ is the probability that each child gets at most $x$ candies. To get $P(X = x)$, we must subtract out the probability that each child gets strictly less than $x$ candies, which is $(\frac{x - 1}{k})^n$.
```

File: content/4_Gold/Combinatorics.mdx [hunk 4]
```diff
@@ -903,49 +920,174 @@ For example, let $X$ be the probability distribution of a fair 6-sided die. $P(X
...
+    expected_max += c * ((c / k) ** n - ((c - 1) / k) ** n)
 
-print("\n".join(ans))
+print(f"{expected_max:.6f}")
+```
+
+</PySection>
+</LanguageSection>
+
+## Linearity of Expectation
```
Comment (ryanchou-dev) [inline content/4_Gold/Combinatorics.mdx:973]: would be cool to link some resources for this section since the explanation is pretty brief.

https://brilliant.org/wiki/linearity-of-expectation/ - off the top of my head, brilliant has an article which is pretty good

File: content/4_Gold/Combinatorics.problems.json [hunk 5]
```diff
@@ -60,6 +60,21 @@
       }
     }
   ],
+  "sample6": [
+    {
+      "uniqueId": "erasing-vertices",
```
Comment (ryanchou-dev) [inline content/4_Gold/Combinatorics.problems.json:65]: fix the unique id

atcoder ids should be the source followed by the problem name: `ac-ErasingVertices`

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
good

**Comment by SansPapyrus683** [comment]
nononon

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
could you use tabs in your solution code?
might also be what's ticking off precommit

**Review by ryanchou-dev — APPROVED** [review]
